server/routerlicious/packages/routerlicious-base/src/utils/sessionHelper.ts

```diff
@@ -269,6 +271,70 @@ function convertSessionToFreshSession(session: ISession, lumberjackProperties):
 	return discoveredNewSession;
 }
 
+export function generateCacheKey(tenantId: string, documentId: string): string {
+	return `session:${tenantId}:${documentId}`;
+}
+
+/**
+ * Get the session from cache if it exists. If it doesn't exist, return undefined.
+ * If the session is "404", return "404" to indicate that the document is deleted.
+ * @internal
+ */
+async function getSessionFromCache(
+	tenantId: string,
+	documentId: string,
+	redisCacheForGetSession: ICache,
+): Promise<ISession | string | undefined> {
+	const session = await redisCacheForGetSession
+		.get(generateCacheKey(tenantId, documentId))
+		.catch((error) => {
+			Lumberjack.error("Error getting session from cache", { tenantId, documentId }, error);
+		});
+	try {
+		if (session) {
+			if (session === "404") {
+				return session;
+			}
+			const sessionObject = JSON.parse(session);
+			return sessionObject as ISession;
+		}
+	} catch (error) {
+		Lumberjack.error("Error parsing session from cache", { tenantId, documentId }, error);
+	}
+	return undefined;
+}
+
+function getEphemeralContainerTtlInSeconds(documentExpirationTime: number): number {
+	// Calculate the TTL based on the document expiration time
+	const currentTime = Date.now();
+	const ttlInSeconds = Math.floor((documentExpirationTime - currentTime) / 1000);
+	return Math.max(ttlInSeconds, 0); // Ensure TTL is not negative
+}
+
+/**
+ * Set the session to cache. If the session is "404", set it as "404" to indicate that the document is deleted.
+ * @internal
+ */
+async function setGetSessionResultInCache(
+	tenantId: string,
+	documentId: string,
+	session: ISession | string,
+	redisCacheForGetSession: ICache,
+	cacheTTLInSeconds: number = defaultGetSessionCacheTtlInSeconds,
+): Promise<void> {
+	const cacheKey = generateCacheKey(tenantId, documentId);
+	try {
+		// If the document is deleted, we set the session as "404" in the cache for a longer time.
+		await redisCacheForGetSession.set(
+			cacheKey,
+			typeof session === "string" && session === "404" ? session : JSON.stringify(session),
+			typeof session === "string" && session === "404" ? 24 * 60 * 60 : cacheTTLInSeconds,
```

Comment:
I think it might be better to reduce this value also to 10 or 20 mins or make it configurable..given the volume of the deleted docs, this can easily fill up redis